### DISCUSSION TITLE
Please explain how pb.filter util works

### DISCUSSION THREAD

**Discussion by laszlo1337** [pocketbase/pocketbase#3889] [Q&A]
According to this section here https://github.com/pocketbase/js-sdk#binding-filter-parameters

the template should automatically handle `null`

but in my experience, when trying something like this:
```js
return pb.collection('orders').getList(page, perPage, {
			filter: pb.filter(
				'(id ~ {:query} || customer.email ~ {:query} || shipping_label_id ~ {:query} || payment_intent_id={:query}) && created >= {:startDate} && created <= {:endDate}',
				{ query, startDate, endDate },
			),
		})
```
when startDate & endDate are null - I will get 0 results. 
If I provide fallback dates:
```
		startDate = startDate ?? new Date(1970, 0, 1)
		endDate = endDate ?? new Date(2050, 0, 1) 
```
then I get expected results.
But reading that readme section I was expecting no filter being applied if these values are null
```
The supported placeholder parameter values are:
- null
```
How else would null be handled?

**Comment by ganigeorgiev** [comment]
If you pass `null` value to a placeholder param it will be inserted as `null`.
I don't understand what is the expected behavior in your case.
You have to provide a fallback value on your own if you don't want `null`, it will not "erase" the expression if that is what you were expecting.

**Reply by laszlo1337** [reply]
Ok, understood. 

That's basically exactly what I was expecting - null means no limit, because if you use null as a limit to date created, then there's **no limit**. Comparing date to null kinda doesn't make sense, at least I think that at the moment but maybe there are some reasons. 
At least this would be a more convienient way of using this helper util, because otherwise you either need to append string if the dates exist or provide arbitrary range fallback.

**Reply by ganigeorgiev** [reply]
> null means no limit,

No, `null` means `null`.

**Reply by lightningspirit** [reply]
@laszlo1337 `null` has semantic meaning in SQL, which makes it not suitable for that use case IMHO. However, in the case of JavaScript, `undefined` really means "not used / defined" which, for SQL, has the same meaning as "not being present".

@ganigeorgiev do you think it could make sense to augment the JS SDK `filter` to include that kind of feature? I know that parsing a string is not easy, but could it make sense to have like a filter "expression builder" using a construction of JS objects like many other ORMs out there?

**Comment by jonshipman** [comment]
Use the fallback dates mate, I usually use 01-01-2100 to guarantee that myself nor anyone I know will be around to fix the problem
